src/saturn_engine/utils/tester/json_utils.py

```diff
@@ -1,5 +1,18 @@
 from typing import Iterator
 
+import dataclasses
+
+
+def normalize_json(obj: object) -> object:
```

Comment:
nit, but you can use [`@overload`](https://mypy.readthedocs.io/en/stable/more_types.html#function-overloading) to make it so it returns a list if it take in a list (And a dict if it takes in a dict). This way you won't have to `cast` thing later on.